src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -53,6 +57,38 @@ public UserResource(UserService userService, LibraryCardService libraryCardServi
         this.gson = new Gson();
     }
 
+    @GET
+    @Produces("application/json")
+    @Path("/role/{roleName}")
+    @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+    public Response getUsers(@Auth AuthUser authUser, @PathParam("roleName") String roleName) {
+        try {
+            User user = userService.findUserByEmail(authUser.getName());
+            if (Objects.nonNull(roleName)) {
+                boolean valid = UserRoles.isValidRole(roleName);
+                if (valid) {
+                    //if there is a valid roleName but it is not SO or Admin then throw an exception
+                    if (!roleName.equals(UserRoles.ADMIN.getRoleName()) && !roleName.equals(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
+                        throw new BadRequestException("Unsupported role name: " + roleName);
+                    }
+                    if (!user.hasUserRole(UserRoles.getUserRoleFromName(roleName))) {
+                        throw new NotFoundException("User: " + user.getDisplayName() + ", " + " does not have " + roleName + " role.");
+                    }
+                    List<User> users = userService.getUsersByUserRole(user, roleName);
+                    return Response.ok().entity(users).build();
+                }
+                else {
+                    throw new BadRequestException("Invalid role name: " + roleName);
+                }
+            } else {
+                throw new BadRequestException("No user role specified.");
+            }
```

Comment:
This block is never executed because `/api/user/role/` is not a valid route. Dropwizard will only hit this method with some kind of value as `roleName`. If you want to handle both `/api/user/role` and `/api/user/role/{roleName}`, you'll need two endpoint methods to do so. Otherwise we get a 404 from DW.

Reply:
Okay makes sense I will remove that block